src/main/main-directive.js

```diff
@@ -2,12 +2,21 @@
     'use strict';
 
     angular.module('stockflux.main')
-        .directive('main', [() => {
+        .directive('main', ['$timeout', ($timeout) => {
             return {
                 restrict: 'E',
                 templateUrl: 'main/main.html',
                 controller: 'MainCtrl',
-                controllerAs: 'mainCtrl'
+                controllerAs: 'mainCtrl',
+                link: (scope, element) => {
+                    scope.$on('compactChanging', () => {
+                        element.addClass('disable-transitions');
+
+                        $timeout(() => {
+                            element.removeClass('disable-transitions');
+                        }, 100);
```

Comment:
I've just tried it, and it doesn't work. Probably something to do with how angular's events are processed.